### PR TITLE
Add all_neighbor_labels function

### PR DESCRIPTION
**Pull request by apriljunge**
Solves #76

### CODE REVIEW

File: test/misc.jl [hunk 1]
```diff
@@ -20,6 +20,8 @@ function test_labels_codes(mg::MetaGraph)
         for label_0 in inneighbor_labels(mg, label_1)
             @test has_edge(mg, code_for(mg, label_0), code_for(mg, label_1))
         end
+        @test collect(all_neighbor_labels(mg, label_1)) ==
+            union(outneighbor_labels(mg, label_1), inneighbor_labels(mg, label_1))
```
Comment (gdalle) [inline test/misc.jl:24]: the union does not guarantee the same order, perhaps sorting first would be more robust

**Review by gdalle — CHANGES_REQUESTED** [review]
thanks for the contribution!

File: test/misc.jl (hunk 1 above)
Comment (bramtayl) [reply test/misc.jl:24]: Using sets could work too

Comment (apriljunge) [reply test/misc.jl]: Thanks for the note. I have changed it so that sets are used.

**Review by gdalle — APPROVED** [review]

**Pull request merged**